addons/crm/models/crm_lead.py

```diff
@@ -0,0 +1,1188 @@
+                date_create = fields.Datetime.from_string(lead.create_date)
+                date_open = fields.Datetime.from_string(lead.date_open)
+                ans = date_open - date_create
+                lead.day_open = abs(int(ans.days))
+
+            if lead.date_closed:
+                date_create = fields.Datetime.from_string(lead.create_date)
+                date_close = fields.Datetime.from_string(lead.date_closed)
+                ans = date_close - date_create
+                lead.day_close = abs(int(ans.days))
+
+    @api.one
+    def _compute_meeting_count(self):
+        self.meeting_count = self.env['calendar.event'].search_count([('opportunity_id', '=', self.id)])
+
+    @api.onchange('partner_id')
+    def on_change_partner_id(self):
+        partner = self.partner_id
+        self.partner_name = (partner.parent_id and partner.parent_id.name) or (partner.is_company and partner.name) or False
+        self.contact_name = (not partner.is_company and partner.name) or False
+        self.title = partner.title and partner.title.id or False
+        self.street = partner.street
+        self.street2 = partner.street2
+        self.city = partner.city
+        self.state_id = partner.state_id and partner.state_id.id or False
+        self.country_id = partner.country_id and partner.country_id.id or False
+        self.email_from = partner.email
+        self.phone = partner.phone
+        self.mobile = partner.mobile
+        self.fax = partner.fax
+        self.zip = partner.zip
+        self.function = partner.function
+
+    @api.onchange('user_id')
+    def on_change_user(self):
+        """ When changing the user, also set a team_id or restrict team id
+            to the ones user_id is member of. """
+        self.team_id = self.env['crm.team']._get_default_team_id()
+
+    @api.onchange('state_id')
+    def onchange_state(self):
+        self.country_id = self.state_id.country_id.id
+
+    @api.onchange('next_activity_id')
+    def onchange_next_activity_id(self):
+        if not self.next_activity_id:
+            self.next_action1 = False
+            self.next_action2 = False
+            self.next_action3 = False
+            self.title_action = False
+            self.date_action = False
+            return
+        activity = self.next_activity_id
+        date_action = False
+        if activity.days:
+            date_action = fields.Datetime.to_string(fields.Datetime.now() + timedelta(days=activity.days))
+        self.next_activity_1 = activity.activity_1_id and activity.activity_1_id.name or False
+        self.next_activity_2 = activity.activity_2_id and activity.activity_2_id.name or False
+        self.next_activity_3 = activity.activity_3_id and activity.activity_3_id.name or False
+        self.title_action = activity.description
+        self.date_action = date_action
+        self.last_activity_id = False
+
+    @api.multi
+    def onchange_stage_id(self, stage_id=False):
```

Comment:
why @api.multi ? called in write method. 
you have to make wrapper class.

   @api.multi
    @api.onchange('stage_id')
    def onchange_stage_id_wrapper(self):
        self.ensure_one()
        values = self.onchange_stage_id(self.stage_id.id). somethinglike this.